library/ConnectionTest/HttpTriggers.cs

```diff
@@ -95,5 +94,21 @@ public static class HttpTriggers
             httpResponseMessage.Content = new StringContent($"caller completed successfully after receiving {linesReceived} lines in {stopwatch.Elapsed}.\n");
             return httpResponseMessage;
         }
+
+        [FunctionName("TestShutdown")]
+        public static async Task TestShutdown(
+           [HttpTrigger(AuthorizationLevel.Anonymous, methods: "get", Route = "testshutdown")] HttpRequestMessage req,
+           CancellationToken token,
+           ILogger log)
+        {
+            try
+            {
+                await Task.Delay(TimeSpan.FromSeconds(300), token);
+            }
+            catch (Exception e)
+            {
+                log.LogError("canceled {e}", e);
+            }
+        }
     }
```

Comment:
How is this helping test for shutdown?

Reply:
I used this to manually validate that `token` is cancelled. Basically calling it and then hitting Ctrl-C in the console while the call is pending.

Reply:
So it is really more of a test of Functions host, not of our code.

I wrote this test because it looked like I was NOT getting cancellation for the HttpRequests. Then I realized that this mechanism only works until we send the http response. So even though we kept the response stream open, the cancellation token would no longer correctly signal the host shutdown. I then found a way to work around this.